src/Domain/FixerFileProcessor.php

```diff
@@ -0,0 +1,190 @@
+        }
+
+        return 'processed';
+    }
+
+    private function processTokensByFixer(SmartFileInfo $smartFileInfo, Tokens $tokens, FixerInterface $fixer): void
+    {
+        try {
+            $fixer->fix($smartFileInfo, $tokens);
+        } catch (\Throwable $throwable) {
+            throw new FixerFailedException(sprintf(
+                'Fixing of "%s" file by "%s" failed: %s in file %s on line %d',
+                $smartFileInfo->getRelativeFilePath(),
+                get_class($fixer),
+                $throwable->getMessage(),
+                $throwable->getFile(),
+                $throwable->getLine()
+            ), $throwable->getCode(), $throwable);
+        }
+
+        if (! $tokens->isChanged()) {
+            return;
+        }
+    }
+
+    private function processDiff(string $diff, string $fixerClass, SmartFileInfo $smartFileInfo): void
+    {
+        $parsedDiff = $this->splitStringByLines($diff);
+        // Get first line number & Remove headers of diff
+        $currentLineNumber = $this->parseLineNumber($parsedDiff[2]);
+        $parsedDiff = array_slice($parsedDiff, 3);
+
+        $headerMessage = "You should change following \n";
+        $currentMessage = $headerMessage;
+        $hasColor = false;
+        foreach ($parsedDiff as $diffLine) {
+            if (mb_strpos($diffLine, '@@ ') === 0) {
+                $this->errorAndDiffCollector->addErrorMessage(
+                    $smartFileInfo,
+                    $currentLineNumber,
+                    $currentMessage,
+                    $fixerClass
+                );
+
+                $currentLineNumber = $this->parseLineNumber($diffLine);
+                $currentMessage = $headerMessage;
+                continue;
+            }
+
+            if (mb_strpos($diffLine, '-') === 0) {
+                $hasColor = true;
+                $currentMessage .= '<fg=red>';
+            }
+            if (mb_strpos($diffLine, '+') === 0) {
+                $hasColor = true;
+                $currentMessage .= '<fg=green>';
+            }
+            $currentMessage .= $diffLine;
+            if ($hasColor) {
+                $hasColor = false;
+                $currentMessage .= '</>';
+            }
+        }
+
+        $this->errorAndDiffCollector->addErrorMessage(
+            $smartFileInfo,
+            $currentLineNumber,
+            $currentMessage,
+            $fixerClass
+        );
+    }
+
+    /**
+     * @param string $input
+     *
+     * @return array<int, string>
+     */
+    private function splitStringByLines(string $input): array
+    {
+        $result = \preg_split('/(.*\R)/', $input, -1, PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY);
+
+        if ($result === false) {
+            throw new \RuntimeException('Unable to split ' . $input);
+        }
+
+        return $result;
+    }
+
+    private function parseLineNumber(string $diffLine): int
+    {
+        $pattern = '@^(?:\@\@ -)?([^,]+)@i';
```

Comment:
For information, lineNumber is formatted like that `@@ -23 +23 @@\n` 

I am not an expert on Regex, so if you have better pattern... :slightly_smiling_face:

Reply:
Actually consider if it's worth using the parser instead :) 
With phpcs fixer there is a parser for diff files, that way the responsibility for parsing the line correctly isn't our responsibility.
https://github.com/PHP-CS-Fixer/diff/blob/master/src/Parser.php

Reply:
@olivernybroe I tried to use the parser you suggested, without success :slightly_frowning_face:  It doesn't return a line number

Reply:
Oh, sorry, did not see that.